### PR TITLE
added custom http client and config.yml

### PR DESCRIPTION
**Pull request by gabriel-dantas98**
- Merged master source repository;
- Increased log messages;
- Added more flexibility custom HTTP client;

### CODE REVIEW

File: abstract.go [hunk 1]
```diff
@@ -39,6 +39,8 @@ func scrapeAwsData(config conf) ([]*tagsData, []*cloudwatchData) {
 
 			resources, metrics := scrapeDiscoveryJob(job, config.Discovery.ExportedTagsOnMetrics, clientTag, clientCloudwatch)
 
+			log.Println("Finished scrap!")
```
Comment (fernandrone) [inline abstract.go]: I think we can remove custom logging for now

File: aws_cloudwatch.go [hunk 2]
```diff
@@ -81,20 +136,16 @@ func createGetMetricStatisticsInput(dimensions []*cloudwatch.Dimension, namespac
...
-				" --dimensions " + dimensionsToCliString(dimensions) +
-				" --namespace " + *namespace +
-				" --statistics " + *statistics[0] +
-				" --period " + strconv.FormatInt(period, 10) +
-				" --start-time " + startTime.Format(time.RFC3339) +
-				" --end-time " + endTime.Format(time.RFC3339))
-		}
-		log.Println(*output)
-	}
+	log.Println("CLI helper - " +
```
Comment (fernandrone) [inline aws_cloudwatch.go:139]: why change this?

File: aws_cloudwatch.go [hunk 3]
```diff
@@ -252,15 +316,24 @@ func getAwsDimensions(job job) (dimensions []*cloudwatch.Dimension) {
 func getMetricsList(dimensions []*cloudwatch.Dimension, serviceName *string, metric metric, clientCloudwatch cloudwatchInterface) (resp *cloudwatch.ListMetricsOutput) {
 	c := clientCloudwatch.client
 	filter := createListMetricsInput(dimensions, getNamespace(serviceName), &metric.Name)
-	req, resp := c.ListMetricsRequest(filter)
-	cloudwatchAPICounter.Inc()
-	err := req.Send()
-
-	if err != nil {
-		panic(err)
+	callListMetrics := false
```
Comment (fernandrone) [inline aws_cloudwatch.go]: a lot of things are changing here?

File: aws_cloudwatch.go [hunk 4]
```diff
@@ -373,21 +446,32 @@ func buildDimension(key string, value string) *cloudwatch.Dimension {
 
 func fixServiceName(serviceName *string, dimensions []*cloudwatch.Dimension) string {
 	var suffixName string
+
 	if *serviceName == "alb" {
+		var albSuffix, tgSuffix string
```
Comment (fernandrone) [inline aws_cloudwatch.go]: here too?

File: prometheus.go [hunk 5]
```diff
@@ -29,8 +29,7 @@ type PrometheusCollector struct {
 
 func NewPrometheusCollector(metrics []*PrometheusMetric) *PrometheusCollector {
 	return &PrometheusCollector{
-		//metrics: removeDuplicatedMetrics(metrics),
-		metrics: metrics,
+		metrics: removeDuplicatedMetrics(metrics),
```
Comment (fernandrone) [inline prometheus.go]: why these changes in this file?

File: main.go [hunk 6]
```diff
@@ -16,7 +16,7 @@ var version = "custom-build"
 var (
 	addr                  = flag.String("listen-address", ":5000", "The address to listen on.")
 	configFile            = flag.String("config.file", "config.yml", "Path to configuration file.")
-	debug                 = flag.Bool("debug", false, "Add verbose logging")
+	debug                 = flag.Bool("debug", true, "Add verbose logging")
```
Comment (fernandrone) [inline main.go]: why change this?

File: aws_cloudwatch.go (hunk 2 above)
Comment (gabriel-dantas98) [reply aws_cloudwatch.go:139]: I was missing some log saying about the metric requests, for some reason this log was not being shown even with the flag enabled. :disappointed:

Comment (gabriel-dantas98) [reply aws_cloudwatch.go:139]: Sample line:
```
2019/10/18 18:13:30 CLI helper - aws cloudwatch get-metric-statistics --metric-name NumberOfMessagesReceived --dimensions Name=QueueName,Value=ProdInsiderCreateReviewDLQ --namespace AWS/SQS --statistics Sum --period 600 --start-time 2019-10-18T18:03:30Z --end-time 2019-10-18T18:13:30Z
```

File: abstract.go (hunk 1 above)
Comment (gabriel-dantas98) [reply abstract.go]: I had added this log to show the end of scrape execution, but I think we can remove it. :smile:

File: main.go (hunk 6 above)
Comment (gabriel-dantas98) [reply main.go]: It was to test, I will correct :wink:

File: aws_tags.go [hunk 7]
```diff
@@ -43,11 +63,30 @@ func createTagSession(region *string, roleArn string) *r.ResourceGroupsTaggingAP
...
+	level := aws.LogDebugWithHTTPBody
+
+	config := &aws.Config{
+		Region: region,
+		HTTPClient: NewHTTPClientWithSettings(HTTPClientSettings{
+			Connect:          55 * time.Second,
+			ExpectContinue:   3 * time.Second,
+			IdleConn:         120 * time.Second,
+			ConnKeepAlive:    30 * time.Second,
+			MaxAllIdleConns:  100,
```
Comment (fernandrone) [inline aws_tags.go:81]: Make MaxAllIdleConns = MaxHostIdleConns, there doesn't seem to be a reason to make them different, since there's only one host.

**Review by fernandrone — CHANGES_REQUESTED** [review]

**Review by fernandrone — APPROVED** [review]
